pennylane/ops/op_math/adjoint.py

```diff
@@ -117,6 +144,12 @@ def circuit(a):
         Adjoint(S)(wires=[0])
 
     """
+    if active_jit := compiler.active_compiler():
+        if lazy is False:
+            raise RuntimeError("Lazy kwarg is not support with qjit.")
```

Comment:
For all Catalyst related errors during program capture or compilation:
```suggestion
            raise CompileError("Lazy kwarg is not support with qjit.")
```